From b5491dad9d2e582536e81f68057ee45388bb55d1 Mon Sep 17 00:00:00 2001
From: Stas Boukarev <stassats@gmail.com>
Date: Thu, 19 Oct 2017 02:04:19 +0300
Subject: [PATCH 200/213] Hardcode the float area uc_mcontext field on arm64.

Glibc keeps playing with ucontexts, hardcoding is the least horrible
solution.

Fixes lp#1724662
---
 src/runtime/arm64-linux-os.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/runtime/arm64-linux-os.c b/src/runtime/arm64-linux-os.c
index 512be72fd..9d4facaa8 100644
--- a/src/runtime/arm64-linux-os.c
+++ b/src/runtime/arm64-linux-os.c
@@ -97,8 +97,11 @@ os_restore_fp_control(os_context_t *context)
 os_context_register_t   *
 os_context_float_register_addr(os_context_t *context, int offset)
 {
+    /* KLUDGE: neither glibc nor the kernel can settle down on a name,
+       the kernel used __reserved, now glibc uses __glibc_reserved1.
+       Hardcode it until they make up their mind */
     return (os_context_register_t*)
-        &((struct fpsimd_context *)context->uc_mcontext.__reserved)->vregs[offset];
+        &((struct fpsimd_context *)((uintptr_t)&context->uc_mcontext + 288))->vregs[offset];
 }
 
 void
-- 
2.13.6

